src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/cas1/Cas1SpaceBookingManagementDomainEventService.kt

```diff
@@ -98,6 +102,61 @@ class Cas1SpaceBookingManagementDomainEventService(
     )
   }
 
+  fun nonArrivalRecorded(
+    user: UserEntity,
+    updatedCas1SpaceBooking: Cas1SpaceBookingEntity,
+    reason: NonArrivalReasonEntity,
+    notes: String?,
+  ) {
+    val domainEventId = UUID.randomUUID()
+    val eventOccurredAt = OffsetDateTime.now().toInstant()
+
+    val application = updatedCas1SpaceBooking.application
+    val premises = updatedCas1SpaceBooking.premises
+    val offenderDetails = getOffenderForCrn(updatedCas1SpaceBooking.crn)
+    val staffUser = getStaffMemberDetails(user.deliusStaffCode)
+
+    domainEventService.savePersonNotArrivedEvent(
+      emit = false,
+      domainEvent = DomainEvent(
+        id = domainEventId,
+        applicationId = application!!.id,
+        crn = updatedCas1SpaceBooking.crn,
+        nomsNumber = offenderDetails?.nomsId,
+        occurredAt = eventOccurredAt,
+        cas1SpaceBookingId = updatedCas1SpaceBooking.id,
+        bookingId = null,
+        data = PersonNotArrivedEnvelope(
+          id = domainEventId,
+          timestamp = eventOccurredAt,
+          eventType = EventType.personNotArrived,
+          eventDetails = PersonNotArrived(
+            applicationId = application!!.id,
+            applicationUrl = applicationUrlTemplate.resolve("id", application.id.toString()),
+            bookingId = updatedCas1SpaceBooking.id,
+            personReference = PersonReference(
+              crn = updatedCas1SpaceBooking.crn,
+              noms = offenderDetails?.nomsId ?: "Unknown NOMS Id",
+            ),
+            deliusEventNumber = application!!.eventNumber,
+            premises = Premises(
+              id = premises.id,
+              name = premises.name,
+              apCode = premises.apCode,
+              legacyApCode = premises.qCode,
+              localAuthorityAreaName = premises.localAuthorityArea!!.name,
+            ),
```

Comment:
not related to this change really, but we do this `ApprovedPremisesEntity` to `Premises` mapping 3 times in this class. Could we pull that out into a private function (or private extension function on the `ApprovedPremisesEntity`?)